lib/controllers/migrate-controller.ts

```diff
@@ -38,15 +40,40 @@ export class MigrateController extends UpdateControllerBase implements IMigrateC
 	];
 
 	private migrationDependencies: IMigrationDependency[] = [
-		{ packageName: constants.TNS_CORE_MODULES_NAME, verifiedVersion: "6.0.0-rc-2019-07-08-111131-01" },
+		{ packageName: constants.TNS_CORE_MODULES_NAME, verifiedVersion: "6.0.0-rc-2019-07-09-183845-06" },
 		{ packageName: constants.TNS_CORE_MODULES_WIDGETS_NAME, verifiedVersion: "6.0.0" },
-		{ packageName: "tns-platform-declarations", isDev: true, verifiedVersion: "6.0.0-rc-2019-06-28-175837-02" },
+		{ packageName: "tns-platform-declarations", isDev: true, verifiedVersion: "6.0.0-rc-2019-07-09-183845-06" },
 		{ packageName: "node-sass", isDev: true, verifiedVersion: "4.12.0" },
-		{ packageName: "typescript", isDev: true, verifiedVersion: "3.4.1" },
+		{
+			packageName: MigrateController.typescriptPackageName, isDev: true, getVerifiedVersion: async (projectData: IProjectData) => {
```

Comment:
extract in a separate private method `getTypescriptVerifiedVersion` for example and move the { packageName: "typescript",...} to the the bottom of the collection to the minimize the chance of somebody adding "@angular/compiler-cli" before it.